region.go

```diff
@@ -287,12 +285,16 @@ func RegionFromZone(zone string) string {
 }
 
 // AvailableSysTypes returns the default system type for the zone.
-func AvailableSysTypes(region string) ([]string, error) {
-	knownRegion, ok := Regions[region]
+func AvailableSysTypes(region, zone string) ([]string, error) {
+	_, ok := Regions[region]
 	if !ok {
 		return nil, fmt.Errorf("unknown region name provided")
 	}
-	return knownRegion.SysTypes, nil
+	knownZone, ok := Regions[region].Zones[zone]
+	if !ok {
+		return nil, fmt.Errorf("unknown zone name provided")
+	}
+	return knownZone, nil
```

Comment:
```suggestion
       knownRegion, ok := Regions[region]
	if !ok {
		return nil, fmt.Errorf("unknown region name provided %s", region)
	}
	supportedSystemTypes, ok := knownRegion.Zones[zone]
	if !ok {
		return nil, fmt.Errorf("unknown zone name provided %s", zone)
	}
	return supportedSystemTypes, nil
```
Would this make it better?

Reply:
Done